TestCombine.playground/Pages/CreatePublisher.xcplaygroundpage/Contents.swift

```diff
@@ -118,3 +118,105 @@ c1?.cancel()
 //}
 //
 //print(c)
+
+public class Observer<Input, Failure: Error> {
+
+    @Synchronized private var isCancelled: Bool = false
+    private let receiveClosure: (Input) -> Subscribers.Demand
+    private let receiveCompletionClosure: (Subscribers.Completion<Failure>) -> Void
+
+    init(receive: @escaping (Input) -> Subscribers.Demand, receiveCompletion: @escaping (Subscribers.Completion<Failure>) -> Void) {
+        receiveClosure = receive
+        receiveCompletionClosure = receiveCompletion
+    }
+
+    public func receive(_ input: Input) -> Subscribers.Demand {
+        guard isCancelled == false else { return .none }
+        return receiveClosure(input)
+    }
+
+    public func receive(completion: Subscribers.Completion<Failure>) -> Void {
+        guard isCancelled == false else { return }
+        return receiveCompletionClosure(completion)
+    }
+
+    fileprivate func cancel() {
+        isCancelled = true
+    }
+}
+
+public struct Orchestrator {
+    public let stop: () -> Void
+    public let start: (Subscribers.Demand) -> Void
+}
+
+public typealias CreationTask<Input, Failure: Error> = (Observer<Input, Failure>, Subscribers.Demand) -> Orchestrator
+
+public struct Create<Output, Failure: Error>: Publisher {
+
+    private let task: CreationTask<Output, Failure>
+    private let demand: Subscribers.Demand
+
+    public init(maxValues: Subscribers.Demand = .unlimited, task: @escaping CreationTask<Output, Failure>) {
+        self.demand = maxValues
+        self.task = task
+    }
+
+    public func receive<S>(subscriber: S) where S : Subscriber, Failure == S.Failure, Output == S.Input {
+        subscriber.receive(subscription: CreateSubscription(publisher: self, subscriber: AnySubscriber(subscriber), maxValues: demand))
+    }
+
+    private class CreateSubscription<Input, Failure: Error>: Subscription {
+
+        @Synchronized private var subscriber: AnySubscriber<Input, Failure>?
```

Comment:
Why @Synchronized?
Maybe cancel must be thread safe?

Reply:
I think it's needed here.
The subscriber variable might became nil at any time during cancel call. `cancel()` vs `receive: { ... }` are likely to get called on different threads.
Am I wrong?

Reply:
I remember cancel() should be called once.
I don't figure out why the same Subscriber/AnyCancellable should be shared among different threads...

P.S.: Ma perché parliamo in inglese? 😂

Reply:
ahaha perché il repo è pubblico, così se vuole intervenire qualcuno sa di cosa parliamo ;)

Reply:
mmm the `AnyCancellable` is kept by an entity requests the sink, a Controller, not a ViewController, for example, might be created on a thread and get used on multiple threads. So multiple threads can try to cancel at the same time transparently accessing the `subscriber` variable from multiple threads.
Isn't this a thread safety related?

Reply:
I understand the scenario but don't know if it makes sense to share an AnyCancellable among different threads.
I would say each thread should make its own subscription on a shared Publisher...

Reply:
It’s also mentioned in the documentation. 

> Subscriptions are class constrained because a Subscription has identity, defined by the moment in time a particular subscriber attached to a publisher. > Canceling a Subscription must be thread-safe.

This means it should be safe to manage it in multithreading environment. Does not matter the thread will cancel it.

Reply:
Is this maybe related with schedulers?